### DISCUSSION TITLE
Crash when saving an entity that has a non nullable property when using Realm code generator without fody

### DISCUSSION THREAD

**Discussion by softlion** [realm/realm-dotnet#3564] [Q&A]
I have 2 curious issues in a Maui net8 app.  
I converted this app to use the Realm source generator.  

1st curious issue: 
- when the realm is created, it does not detect any entity. 
I worked around by specifying the schema explicitely.

2nd issue (unresolved):
- realm crashes when adding a new object to a table. The crash message explains that it is unable to set a null value for a non nullable property. 
I verified that the object being saved has only non null properties.  
I checked the generated source code, but can't find what could be wrong (can't put a breakpoint there).

Any clue ?

**Comment by nirinchev** [comment]
Hey @softlion, do you think you can provide us with a repro case for this?

**Reply by softlion** [reply]
https://github.com/softlion/RealmBug

**Reply by softlion** [reply]
I want to use only the Realm generator (without fody), is that possible ?

**Reply by papafe** [reply]
Thanks for your repro case. No, at the moment you can't use only the Source Generator, because we still need to weave the setters and getters of realm objects.

**Reply by papafe** [reply]
@softlion I just run your example
- I can see you're not specifying the schema explicitly, so I suppose you don't have that issue anymore?
- Regarding the crash. Your `Name` property is not declared as nullable (like `Alias` is), so when you try to add the object to the realm you get the exception. If you set a value for name when you create the `DbWallet` it will work

**Reply by softlion** [reply]
> Regarding the crash. Your Name property is not declared as nullable (like Alias is), so when you try to add the object to the realm you get the exception. If you set a value for name when you create the DbWallet it will work

Well that's the issue: the Name property I'm saving is NEVER null.

**Reply by papafe** [reply]
When you're creating the `DbWallet` ([here](https://github.com/softlion/RealmBug/blob/0630809ec6083dfdf465cd63bb12707028482fa7/App.xaml.cs#L33)), you're setting only the `WalletId`, so the `Name` property is null
